app/src/main/java/org/dhis2/usescases/login/LoginActivity.kt

```diff
@@ -305,8 +304,8 @@ class LoginActivity : ActivityGlobalAbstract(), LoginContracts.View {
         binding.serverUrlEdit.dropDownWidth = resources.displayMetrics.widthPixels
         binding.userNameEdit.dropDownWidth = resources.displayMetrics.widthPixels
 
-        urls = getListFromPreference(Constants.PREFS_URLS)
-        users = getListFromPreference(Constants.PREFS_USERS)
+        urls = presenter.preferenceProvider.getSet(Constants.PREFS_URLS, emptySet())!!.toMutableList()
```

Comment:
A few things :)
Views should not know about datastores, therefore we should not expose preferencesProvider as public.
The way this should go is :
view calls presenter, for ej: presenter.renderURL(), presenter fetch from preferences provider and then send the results to the view